Allow the node to reattach with different composition data

jakubwitowski-silvair opened the pull request.

michallowasrzechonek-silvair committed:
mesh: Add mesh-io for Silvair Radio over UART or TCP

michallowasrzechonek-silvair committed:
mesh: Check that incoming packet fits in net key cache

michallowasrzechonek-silvair committed:
mesh: Replace aes_cmac with aes_cmac_one

michallowasrzechonek-silvair committed:
mesh: Add option to use openssl for crypto

michallowasrzechonek-silvair committed:
mesh: Remove redundant code from mesh/crypto

- Remove application_encrypt/application_decrypt
- Make mesh_crypto_privacy_counter private, add mesh_crypto_pecb
- Make *_nonce functions private and align their implementation to be
  more consistent
- Refactor network_encrypt/network_decrypt to use *_nonce functions and
  rename them to packet_encrypt/packet_decrypt
- Refactor packet_encode/packet_decode

jakubwitowski-silvair committed:
mesh: Add DBus method for updating sequence number

michallowasrzechonek-silvair committed:
mesh: Revert "Base config json code on newer version of json-c"

This reverts commit 11d7fe8535be45bb354b71cd6db0b394b3f178d7.

Package for json-c 0.13 is not available on Ubuntu 18.10, revert this
until we upgrade.

jakubwitowski-silvair committed:
mesh: add cid/pid/vid setter

This patch adds the setter for the CID PID and VID.

michallowasrzechonek-silvair commented on mesh/node.c:
```diff
@@ -710,6 +710,19 @@ bool node_default_ttl_set(struct mesh_node *node, uint8_t ttl)
 
+bool node_composition_data_set(struct mesh_node *node)
+{
+	struct node_composition *comp;
```
Remove local alias, use `node->comp->...` directly

jakubwitowski-silvair committed:
mesh: allow to reattach with new composition data

This patch allows to change the CIP/PID/VID data when
reattaching. Additionally the device key is verified
during those change.

michallowasrzechonek-silvair commented on mesh/node.c:
```diff
@@ -1406,11 +1416,27 @@ static bool check_req_node(struct managed_obj_request *req)
 		node_comp[8] = 0;
 		attach_comp[8] = 0;
 
```
`memcpy(attach_comp, node_comp, ...)`

przemyslawfierek-silvair committed:
mesh: Add support for 'io_set_write_handler' to the 'silvair-io'.

michallowasrzechonek-silvair commented:
Brian would like this to check if node's device key exists in the keyring. Note that we need to check key *value*, not only the fact it exists.

michallowasrzechonek-silvair commented on mesh/node.c:
```diff
@@ -1402,15 +1392,52 @@ static bool check_req_node(struct managed_obj_request *req)
+		keyring_get_remote_dev_key(req->attach, req->attach->primary,
+							keyring_dev_key);
+
+		if (memcmp(keyring_dev_key, storage_dev_key, 16))
```
Remove local variable, `memcmp(node_get_device_key(req->attach), storage_dev_key, 16)`

michallowasrzechonek-silvair commented on mesh/node.c:
```diff
@@ -1402,15 +1392,52 @@ static bool check_req_node(struct managed_obj_request *req)
+		if (!memcmp(node_comp, attach_comp, COMP_ID_SIZE))
+			return true;
+
+		l_debug("COMP ID update");
```
Remove log

michallowasrzechonek-silvair commented:
Way better! Two cosmetics and LGTM.

michallowasrzechonek-silvair commented:
Superseded by 8a382262125787caf38a1f800ec8956a1bf71d85 and cfdc59d372c070d53b9180066c634af61c8fa170
